tls/data_source_public_key_test.go

```diff
@@ -0,0 +1,42 @@
+package tls
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+const (
+	expectedPublic = `-----BEGIN PUBLIC KEY-----
+MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQDPLaq43D9C596ko9yQipWUf2Fb
+RhFs18D3wBDBqXLIoP7W3rm5S292/JiNPa+mX76IYFF416zTBGG9J5w4d4VFrROn
+8IuMWqHgdXsCUf2szN7EnJcVBsBzTxxWqz4DjX315vbm/PFOLlKzC0Ngs4h1iDiC
+D9Hk2MajZuFnJiqj1QIDAQAB
+-----END PUBLIC KEY-----`
+	expectedPublicSSH = `ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAAAgQDPLaq43D9C596ko9yQipWUf2FbRhFs18D3wBDBqXLIoP7W3rm5S292/JiNPa+mX76IYFF416zTBGG9J5w4d4VFrROn8IuMWqHgdXsCUf2szN7EnJcVBsBzTxxWqz4DjX315vbm/PFOLlKzC0Ngs4h1iDiCD9Hk2MajZuFnJiqj1Q==`
+)
+
+func TestAccPublicKey_dataSource(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(testAccDataSourcePublicKeyConfig, testPrivateKey),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("data.tls_public_key.test", "public_key_pem", strings.TrimSpace(expectedPublic)+"\n"),
+					resource.TestCheckResourceAttr("data.tls_public_key.test", "public_key_openssh", strings.TrimSpace(expectedPublicSSH)+"\n"),
+				),
+			},
+		},
+	})
+}
+
+const testAccDataSourcePublicKeyConfig = `
+data "tls_public_key" "test" {
```

Comment:
PS: You could probably make this test self-hosting but using the `tls_private_key` resource to generate a key. Then you could use [`TestCheckResourceAttrPair`](https://github.com/hashicorp/terraform/blob/master/helper/resource/testing.go#L1016) to just compare the two values.

Reply:
I'll add it as another test. I like the fact that we're taking a private key without it's matching Public Key and confirming that its output is what we expected.

Reply:
@FrenchBen fair enough, and I actually didn't notice the `testPrivateKey` fixture being used in other tests. Will wait for the second test!